documentation/blog/2025-10-03-running-goose-in-containerized-environments/index.md

```diff
@@ -184,78 +183,141 @@ analyze:
-docker run --rm --network host goose:local
+# Allow goose to access your local services
+$ docker run --rm --network host \
+   --env-file .env \
+   -v $(pwd):/workspace \
+   ghcr.io/block/goose:v0.9.3 run -t "Test the API endpoints in our local development server"
 ```
 
+**Security note:** Only use host networking in development environments. For production, use proper service discovery and networking configurations.
+
 ## Advanced Patterns
 
 ### Resource Limits
 
 Set memory and CPU limits for production:
 
 ```bash
-docker run --rm \
-    --memory="2g" \
-    --cpus="2" \
-    goose:local
+$ docker run --rm \
+   --memory="2g" \
+   --cpus="2" \
+   goose:local
+```
+
+### Debugging Container Issues
+
+When goose behaves unexpectedly in containers, you need to investigate the environment. Here are common debugging scenarios:
+
+**Inspect the container environment:**
+
+```bash
+# Drop into a shell to examine the container
+$ docker run --rm -it --entrypoint bash \
+   -v $(pwd):/workspace \
+   --env-file .env \
+   ghcr.io/block/goose:v0.9.3
+
+# Inside the container, check:
+goose@container:~$ env | grep GOOSE      # Environment variables
+goose@container:~$ goose --version       # Binary version
+goose@container:~$ ls -la /workspace     # File permissions
+goose@container:~$ cat ~/.config/goose/config.yaml  # Configuration
 ```
 
-### Custom Entrypoints
+**Debug API connectivity issues:**
+
+```bash
+# Test network connectivity and API access
+$ docker run --rm -it --entrypoint bash goose:v0.9.3
+goose@container:~$ curl -I https://api.openai.com/v1/models
+goose@container:~$ goose configure --check  # Verify API keys
+```
 
-Need to debug? Override the entrypoint:
+**Examine goose logs in verbose mode:**
 
 ```bash
-docker run --rm -it --entrypoint bash goose:local
+$ docker run --rm -it \
+   -v $(pwd):/workspace \
+   --env-file .env \
+   ghcr.io/block/goose:v0.9.3 \
+   --verbose run -t "Your failing command"
 ```
 
+This debugging approach helps identify issues with file permissions, network connectivity, API authentication, or configuration problems that might not be obvious from normal goose output.
+
 ### Multi-Platform Builds
 
 For deployment across architectures:
 
 ```bash
-docker buildx build --platform linux/amd64,linux/arm64 -t goose:multi .
+$ docker buildx build --platform linux/amd64,linux/arm64 -t goose:multi .
 ```
 
 ## Production Considerations
 
-For production deployments:
-
-1. **Use specific tags**: Avoid `latest` for reproducibility
+For production deployments, replace all the `latest` tags from our examples with specific release versions:
 
-2. **Secrets management**: Use Docker secrets or external secret stores for API keys
+### Use Specific Release Tags
 
-3. **Logging**: Configure log aggregation
+```bash
+# Instead of this (unpredictable):
+$ docker pull ghcr.io/block/goose:latest
 
-4. **Monitoring**: Set up health checks and metrics
+# Use this (reproducible):
+$ docker pull ghcr.io/block/goose:v0.9.3
```

Comment:
can we set this to a modern version tag like 1.9.1 or something? 0.9.3 was pretty old :)